src/Form/Settings.php

```diff
@@ -0,0 +1,131 @@
+   *
+   * @var \Drupal\Core\Config\Entity\ConfigEntityStorageInterface
+   */
+  protected $passwordConstraintEntityStorage;
+
+  /**
+   * Constructs the settings from for password constraint.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   Config factory.
+   * @param \Drupal\Core\Config\Entity\ConfigEntityStorageInterface $password_constraint_entity_storage
+   *   Password constraint entity storage.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   Messenger.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, ConfigEntityStorageInterface $password_constraint_entity_storage, MessengerInterface $messenger) {
+    parent::__construct($config_factory);
+
+    $this->passwordConstraintEntityStorage = $password_constraint_entity_storage;
+    $this->messenger = $messenger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): Settings {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('entity_type.manager')->getStorage('password_enhancements_constraint'),
+      $container->get('messenger')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'password_enhancements_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames(): array {
+    return [
+      'password_enhancements.settings',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    if ($this->config('user.settings')->get('password_strength')) {
+      $this->messenger->addWarning($this->t('The core <a href="@url" target="_blank">password strength indicator</a> is enabled, it is recommended to disable it.', [
+        '@url' => Url::fromRoute('entity.user.admin_form', [], [
+          'fragment' => 'edit-user-password-strength',
+        ])->toString(),
+      ]));
+    }
+
+    $password_settings = $this->config('password_enhancements.settings');
+
+    $form['constraint_update_effect'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Constraint update effect'),
+      '#description' => $this->t('Sets the update effect for password constraint completion.'),
+      '#options' => [
+        static::CONSTRAINT_EFFECT_STRIKETHROUGH => $this->t('Strike-through'),
+        static::CONSTRAINT_EFFECT_HIDE => $this->t('Hide'),
+      ],
+      '#default_value' => $password_settings->get('constraint_update_effect') ?? static::CONSTRAINT_EFFECT_STRIKETHROUGH,
+    ];
+
+    $form['require_password'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Always require password at registration'),
+      '#description' => $this->t('If enabled the password field will be always visible and required on the registration form.'),
+      '#default_value' => $password_settings->get('require_password') ?? FALSE,
```

Comment:
Define a default value for this in `config/install/password_enhancements.settings.yml` and you won't need to check whether it is null.